src/bin/pgcopydb/catalog.c

```diff
@@ -6487,6 +6487,66 @@ catalog_add_s_extension_config(DatabaseCatalog *catalog,
 }
 
 
+/*
+ *   catalog_iter_s_extesnion_timescaledb_checker , iterates over the list of extensions in our catalogs
+ *   and checks for presence of timescaledb extension.
+ */
+bool
+catalog_has_timescaledb_extension(DatabaseCatalog *catalog,
+											 bool *timescaledb)
+{
```

Comment:
The new name needs updating in the comment also. But the problem that I have with that code is not even with the function name. Why use an iterator over the full list of extensions in our SQLite catalogs when searching for the presence of only one, that we know the name of? This should be `SELECT 1 FROM s_extensions WHERE extname = $1`, like I said before: a **lookup** function.

Reply:
We have created a lookup function as you asked. It now uses `SELECT 1 FROM s_extensions WHERE extname = $1` to efficiently check for the presence of an extension.